agent/consul/state/config_entry_test.go

```diff
@@ -3714,3 +3714,132 @@ func TestStateStore_DiscoveryChain_AttachVirtualIPs(t *testing.T) {
 	require.Equal(t, []string{"2.2.2.2", "3.3.3.3"}, chain.ManualVirtualIPs)
 
 }
+
+func TestCollectJWTProviderNames(t *testing.T) {
```

Comment:
This is testing an implementation detail and not the actual logic of  `validateJWTProviderIsReferenced`. Given that this is in the `state` package, could we test with a mock state store and try deletes from that layer?